File: solutions/typescript/roman-numerals/1/roman-numerals.ts
export const toRoman = (num: number): string => {
  const map: { [key: number]: string } = {
    1000: 'M', 900: 'CM', 500: 'D', 400: 'CD',
    100: 'C', 90: 'XC', 50: 'L', 40: 'XL',
    10: 'X', 9: 'IX', 5: 'V', 4: 'IV', 1: 'I'
  };

  let result = '';
  const values = Object.keys(map).map(Number).sort((a, b) => b - a);

  for (const value of values) {
    while (num >= value) {
      result += map[value];
      num -= value;
    }
  }

  return result;
};

File: solutions/kotlin/roman-numerals/1/src/main/kotlin/RomanNumerals.kt
object RomanNumerals {

    fun value(n: Int): String {
        var number = n
        val result = StringBuilder()
        
    
        
        val arabicValues = intArrayOf(1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1)
        val romanSymbols = arrayOf("M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I")
        
   
        
        for (i in arabicValues.indices) {
            while (number >= arabicValues[i]) {
                result.append(romanSymbols[i]) 
                
                number -= arabicValues[i]      
                
            }
        }
        
        return result.toString()
    }
}
